flink-ml-lib/src/test/java/org/apache/flink/ml/classification/NaiveBayesTest.java

```diff
@@ -0,0 +1,297 @@
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0)
+                };
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testInputNotContainLabel() throws Exception {
+        errorMessage =
+                "Naive Bayes should throw exception if input table schema does not contain label column.";
+        labelCol = "non-label";
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testNullPredict() throws Exception {
+        errorMessage = "Naive Bayes should throw exception if predict col is not set.";
+        predictCol = null;
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testNullModelType() throws Exception {
+        errorMessage = "Naive Bayes should throw exception if model type is not set.";
+        modelType = null;
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testVectorWithDiffLen() throws Exception {
+        errorMessage =
+                "Naive Bayes should throw exception if length of feature vectors are not equal.";
+        trainData =
+                new Row[] {
+                    Row.of(1., Vectors.dense(1, 1., 1., 1.), 11.0),
+                    Row.of(1., Vectors.dense(1, 1., 0., 1., 2.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1., 3.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.5, 2.), 11.0),
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0)
+                };
+        runAndCheck();
+    }
+
+    @Test
+    public void testSaveLoad() throws Exception {
+        errorMessage = "Naive Bayes should be able to save Model to filesystem and load correctly.";
+        isSaveLoad = true;
+        runAndCheck();
+    }
+
+    private void runAndCheck() throws Exception {
+        Table trainTable =
+                tEnv.fromDataStream(
+                                env.fromElements(trainData)
+                                        .assignTimestampsAndWatermarks(
+                                                WatermarkStrategy.noWatermarks()),
+                                schema)
+                        .as("weight", "features", "label");
+        Table predictTable =
+                tEnv.fromDataStream(
+                                env.fromElements(predictData)
+                                        .assignTimestampsAndWatermarks(
+                                                WatermarkStrategy.noWatermarks()),
+                                schema)
+                        .as("weight", "features", "label");
+
+        NaiveBayes estimator = new NaiveBayes();
```

Comment:
It is preferred to follow best practice when writing test code, since it is what users will read and probably copy.

We can use `NaiveBayes estimator = new NaiveBayes().setSmoothing(smoothing)` here.